identity-apps-core/apps/x509-certificate-authentication-portal/src/main/webapp/includes/branding-preferences.jsp

```diff
@@ -0,0 +1,669 @@
+        // Check cookie for the user selected language first.
+        Cookie[] cookies = request.getCookies();
+
+        // Map to store default supported language codes.
+        Map<String, String> supportedLanguages = new HashMap<>();
+        supportedLanguages.put("en", "US");
+        supportedLanguages.put("fr", "FR");
+        supportedLanguages.put("es", "ES");
+        supportedLanguages.put("pt", "PT");
+        supportedLanguages.put("de", "DE");
+        supportedLanguages.put("zh", "CN");
+        supportedLanguages.put("ja", "JP");
+
+        List<String> languageSupportedCountries = new ArrayList<>();
+        languageSupportedCountries.add("US");
+        languageSupportedCountries.add("FR");
+        languageSupportedCountries.add("ES");
+        languageSupportedCountries.add("PT");
+        languageSupportedCountries.add("DE");
+        languageSupportedCountries.add("CN");
+        languageSupportedCountries.add("JP");
+        languageSupportedCountries.add("BR");
+
+        if (cookies != null) {
+            for (Cookie cookie : cookies) {
+                if (cookie.getName().equals(COOKIE_NAME)) {
+                    localeFromCookie = cookie.getValue();
+                }
+            }
+        }
+        // Set lang from the priority order.
+        if (localeFromCookie != null) {
+            lang = localeFromCookie;
+            try {
+                String langStr = "en";
+                String langLocale = "US";
+
+                if (lang.contains("_")) {
+                    langStr = lang.split("_")[0];
+                    langLocale = lang.split("_")[1];
+                } else if (lang.contains("-")) {
+                    langStr = lang.split("-")[0];
+                    langLocale = lang.split("-")[1];
+                }
+
+                userLocale = new Locale(langStr, langLocale);
+            } catch (Exception e) {
+                // In case the language is defined but not in the correct format.
+                userLocale = browserLocale;
+            }
+        } else if (uiLocaleFromURL != null) {
+            for (String localeStr : uiLocaleFromURL.split(" ")) {
+                String langStr = "en";
+                String langLocale = "US";
+
+                if (lang.contains("_")) {
+                    langStr = lang.split("_")[0];
+                    langLocale = lang.split("_")[1];
+                } else if (lang.contains("-")) {
+                    langStr = lang.split("-")[0];
+                    langLocale = lang.split("-")[1];
+                }
+
+                Locale tempLocale = new Locale(langStr, langLocale);
+                // Trying to find out whether we have a resource bundle for the given locale.
+                try {
+                    ResourceBundle foundBundle = ResourceBundle.getBundle(BUNDLE, tempLocale);
+                    // If so, setting the userLocale to that locale. If not, set the browser locale as user locale
```

Comment:
How about rewording the comment as follows?

`If the i18n bundle for the locale is found, set the userLocale to that locale...`

Reply:
https://github.com/wso2/identity-apps/pull/6251/commits/fc1f823815fb2c79f4053a0732318067fbc830f6